src/turing_model.jl

```diff
@@ -90,11 +100,57 @@ function turing_model(
     if standardize
         μ_X, σ_X, X = standardize_predictors(X)
         μ_y, σ_y, y = standardize_predictors(y)
+        if !isnothing(Z)
+            #TODO: implement random-effects slope
+            throw(
+                ArgumentError(
+                    "TuringGLM currently does not support random-effects for slope terms"
+                ),
+            )
+        end
     end
 
     # Random-Effects Conditionals
     if has_ranef(formula)
-        # TODO
+        if priors isa DefaultPrior
+            custom_prior = CustomPrior(
+                TDist(3), LocationScale(median(y), mad(y), TDist(3)), nothing
+            )
+        else
+            custom_prior = priors
+        end
+        intercept_ranef = intercept_per_ranef(ranef(formula))
+        group_var = first(ranef(formula)).rhs
+        idx = get_idx(term(group_var), data)
+        # print for the user the idx
+        println("The idx are $(last(idx))\n")
+        @model function normal_model_ranef(
+            y,
+            X;
+            predictors=size(X, 2),
+            idxs=first(idx),
+            n_gr=length(unique(first(idx))),
+            intercept_ranef=intercept_ranef,
+            μ_X=μ_X,
+            σ_X=σ_X,
+            prior=custom_prior,
+            residual=1 / std(y),
+        )
+            α ~ prior.intercept
+            β ~ filldist(prior.predictors, predictors)
+            σ ~ Exponential(residual)
+            μ = α .+ X * β
+            if !isempty(intercept_ranef)
+                τ ~ LocationScale(0, mad(y), truncated(TDist(3), 0, Inf))
```

Comment:
FYI. This is very soon going to be AffineDistribution: https://github.com/JuliaStats/Distributions.jl/pull/1453/.

Reply:
Good to know! Had no Idea!